### ISSUE TITLE
baseUrl is duplicated in file paths checking for file existence

### ISSUE THREAD

**Issue by Atlinx** [benyap/resolve-tspaths#73]
When I was using `resolve-tspaths`, I had my baseUrl set to `src`. This produced filepaths like `absolute\path\src\src\app\config`, where the `src` directory gets incorrectly doubled in the filepath. Digging deeper, I found that the aliases were being stored relative to the baseUrl (ie. `src\app\controllers\*`). However, when actually used, they were added to baseUrl file path (ie. `absolute\path\src`).

**Comment by benyap** [comment]
Hi @Atlinx, I could not reproduce this error. Could you provide a sample `tsconfig.json` file that produces this error?

Another thing to check - with your `baseUrl` to `src`, did you make sure that your path aliases were relative to `baseUrl`?

**Comment by Atlinx** [comment]
Here are the tsconfig files I used

tsconfig.json
```JSON
{
	"compilerOptions": {
		"module": "commonjs",
		"esModuleInterop": true,
		"target": "es6",
		"noImplicitAny": true,
		"moduleResolution": "node",
		"sourceMap": true,
		"outDir": "dist",
		"baseUrl": "src",
		"paths": {
			"*": [
				"../node_modules/*"
			],
			"@app/*": [ "app/*" ],
			"@config/*": [ "app/config/*" ],
			"@models/*": [ "app/models/*" ],
			"@utils/*": [ "app/utils/*" ],
			"@routes/*": [ "app/routes/*" ],
			"@controllers/*": [ "app/controllers/*" ],
			"@middlewares/*": [ "app/middlewares/*" ],
			"@init/*": [ "app/init/*" ],
			"@tests/*": [ "tests/*" ]
		},
		/* Advanced Options */
		"plugins": [
			{ "transform": "typescript-transform-paths" },
			{ "transform": "typescript-is/lib/transform-inline/transformer" },
			{ "transform": "ts-transformer-keys/transformer" }
		],
	},
	"include": [
		"**/*"
	],
}
```

tsconfig.build.json
```JSON
{
  "extends": "./tsconfig.json",
  "exclude": [
    "**/*.spec.ts",
    "**/*.fixture.ts",
    "/tests"
  ]
}
```

**Comment by benyap** [comment]
Hi @Atlinx, thanks for providing that. I set up a local repo with your configuration and I still can't seem to reproduce your issue. A few questions:

* Which version of `resolve-tspaths` are you using? If it's not the latest version, could you upgrade and give that a shot?
* If it's still an issue, can you also provide a minimal repo that reproduces the issue you're having?

**Comment by Atlinx** [comment]
@benyap Here's a minimum repro:
[resolve-tspaths-repro.zip](https://github.com/benyap/resolve-tspaths/files/8746965/resolve-tspaths-repro.zip)

Open it and run `npm start`.

**Comment by benyap** [comment]
Hi @Atlinx, thanks for that! I think I've found the solution to the issue you're having.

But first - just thought I'd mention that in your reproduction package.json, you're not actually transforming your output correctly. You need to run `resolve-tspaths` **after** you've built your code with `tsc`, i.e. your `build` script should be `npm run fixpaths && tsc`, not the other way around.

Back to your issue - in most projects, the source code will usually live inside a `src` folder. `resolve-tspaths` makes this assumption, which is documented [here](https://github.com/benyap/resolve-tspaths#options), as it is the most common use case. However, in your example, you have source files located in the **root** of your directory instead of a `src` folder, which led to unexpected behaviour when you tried to transform the paths in your build output.

In your case, your source directory is actually `.` instead of `src`. To override the default, change the `fixpaths` script in your example to be `resolve-tspaths --src .`, and you should see that it actually will transform your output correctly. My guess is that in your original post, you probably had a `src` directory, but also had files outside of that, making your actual source directory `.`. When you used `tsc` to build your project, it would have created files with a `src` folder inside your `dist` directory. So that, along with the default `--src` being `src`, would have led to paths with `src/src` like you mentioned.

**So in summary - please make sure you specify an appropriate `--src` option if your source code isn't in a `src` directory.**

I'm going to close this issue for now, and I'll update the documentation to make this a little clearer. If you are still having problems, please feel free to leave a comment or open another issue!

**Comment by Atlinx** [comment]
Would it be possible to use the "rootDir" option in the tsconfig file as the default "src" argument?

**Comment by benyap** [comment]
@Atlinx this has been released in [v0.5.0](https://github.com/benyap/resolve-tspaths/releases/tag/v0.5.0). Thanks for your suggestion!
